third_party/blink/renderer/core/inspector/inspector_dom_agent.cc

```diff
@@ -1057,8 +1061,9 @@ Response InspectorDOMAgent::performSearch(
     Maybe<bool> optional_include_user_agent_shadow_dom,
     String* search_id,
     int* result_count) {
-  if (!enabled_.Get())
-    return Response::ServerError("DOM agent is not enabled");
+  // [replay] we cannot currently enable these during replay
+  // if (!enabled_.Get())
+  //   return Response::ServerError("DOM agent is not enabled");
```

Comment:
ditto

Reply:
Similarly a recordreplay::HasDivergedFromRecording call can be used here to avoid the error path when we're getting the DOM state while paused when replaying.

Reply:
Does all pause command handler code have `HasDivergedFromRecording` true?

Reply:
Within the backend, any pauseRequest sent to a pause will cause it to diverge before sending back an answer (see PauseRequestHandler in Manifest.cpp).  From the application's perspective though some commands may be sent while not diverged, e.g. GetConsoleMessageData (which is called when not diverged) can get the stack contents and object previews.  All DOM related commands will be sent while diverged, but this part of the system is underspecified and it would be good to have a better defined contract between the recorder and the application.